tinygrad/renderer/triton.py

```diff
@@ -0,0 +1,134 @@
+
+#NOTE triton requires matching dimensions for load/store, disable this and see TestOps::test_output_padded_conv_transpose2d fail to compile
+def fill_dims_for_idx(idx, dims):
+  return "(" + idx.render() + "+ (" + (f"0 * ({'+'.join(d for d in dims)})))")
+
+def get_max(var):
+  if isinstance(var.max, int): return var.max
+  return re.sub(r'\[(.*?)\]', '', str(var.max))[1:-1]
+
+#NOTE can be removed after https://github.com/gpuocelot/gpuocelot/issues/8 gets resolved
+def remove_single_scalar_curly_braces(ptx_code):
+  return '\n'.join([re.sub(r'\{\s*(%\w+)\s*\}', r'\1', line) for line in ptx_code.split('\n')])
+
+def define_scalar(local_size, triton_type, args):
+  val = (('-' if args<0 else '') + 'float("inf")') if math.isinf(args) else ('float("nan")' if math.isnan(args) else str(args))
+  if len(local_size) > 0:
+    return f"tl.full(({','.join([str(next_power_of_2(x)) for x in local_size])},),{val}, dtype={triton_type})"
+  return f"tl.where(1, {val}, {val}).to({triton_type})"
+
+def uops_to_triton(function_name:str, uops:List[UOp]):
+  kernel = []
+  global_size: List[int] = []
+  local_size: List[int] = []
+  depth = 1
+  bufs = []
+  signatures = []
+  dims = []
+  c: DefaultDict[str, int] = defaultdict(int)
+  def ssa(prefix="t"):
+    nonlocal c
+    c[prefix] += 1
+    return f"{prefix}{c[prefix]-1}"
+  r: Dict[UOp, str] = {}
+  def kk(s): kernel.append("  "*depth+s)  
+  gid = [f"tl.program_id({i})" for i in range(3)]
+  code_for_op: Final[Dict[Op, Callable]] = {
+    UnaryOps.EXP2: lambda x: f"tl.math.exp2({x})",
+    UnaryOps.LOG2: lambda x: f"tl.math.log2({x})", # TODO: is fast_log2f ok?
+    UnaryOps.SIN: lambda x: f"tl.sin({x})",
+    UnaryOps.SQRT: lambda x: f"tl.sqrt({x})",
+    UnaryOps.NEG: lambda x: f"-{x}",
+    BinaryOps.ADD: lambda x,y: f"({x}+{y})", BinaryOps.SUB: lambda x,y: f"({x}-{y})",
+    BinaryOps.MUL: lambda x,y: f"({x}*{y})", BinaryOps.DIV: lambda x,y: f"({x}/{y})",
+    BinaryOps.MAX: lambda x,y: f"tl.maximum({x},{y})", # axis?
+    BinaryOps.CMPLT: lambda x,y: f"({x}<{y})",
+    BinaryOps.MOD: lambda x,y: f"({x}%{y})",
+    TernaryOps.MULACC: lambda x,y,z: f"(({x}*{y})+{z})",
+    TernaryOps.WHERE: lambda x,y,z: f"tl.where({x},{y},{z})",
+  }
+  triton_dtypes = {dtypes.double: "tl.float64", dtypes.float32: "tl.float32", dtypes.float16: "tl.float16", dtypes.bool: "tl.int8", dtypes.int8: "tl.int8", dtypes.uint8: "tl.uint8", dtypes.int32: "tl.int32", dtypes.int64: "tl.int64"}
+  signature_dtypes = {dtypes.double: "*fp64",dtypes.float32: "*fp32", dtypes.float16: "*fp16", dtypes.bool: "*i8", dtypes.int8: "*i8", dtypes.uint8: "*u8", dtypes._arg_int32: "i32", dtypes.int32: "*i32", dtypes.int64: "*i64"}
+
+  for u in uops:
+    uop,newvar,vin,args,_ = u
+    if uop == UOps.LOOP:
+      for i,var in enumerate(args[0]):
+        if isinstance(var, NumNode): continue # python doesnt have block scope
+        else:
```

Comment:
Else can be removed since if block contains continue

Reply:
done